### DISCUSSION TITLE
Korean language

### DISCUSSION THREAD

**Discussion by yseo1110** [devemux86/cruiser#402] [Q&A] [answered]
I'd like to translate into Korean language for Korean users here.
How can I translate Cruiser?

**Comment by devemux86** [comment]
@yseo1110 thanks for your offer!

Korean is one of my favorite languages, although I'm still learning it! :slightly_smiling_face: 

Cruiser texts are located in several text files. I will attach them here so you can translate them.

**Reply by yseo1110** [reply]
Thanks for your kind response.
How can I find text file of Cruiser for translation?

**Reply by devemux86** [reply]
@yseo1110 please give me some time to collect the texts, as new content is added frequently.

I will attach the text files here in a while.

**Reply by yseo1110** [reply]
We have our own made Korea DEM file, but cannot use it.
If you can add DEM menu so that we can load DEM file, it will be very useful.

**Reply by devemux86** [reply]
@yseo1110 do you mean DEM with elevation data?
And we still need a map to use dem files?

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 19:12 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> please give me some time to
> collect the texts, as new content is added frequently.
>
> I will attach the text files here in a while.
>
If you have Google spreadsheet for translation, it will be very helpful.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 19:12 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> please give me some time to
> collect the texts, as new content is added frequently.
>
> I will attach the text files here in a while.
>

Is it possible to load Offline map routing with your latest version?

**Reply by devemux86** [reply]
@yseo1110 

> Is it possible to load Offline map routing with your latest version?

Cruiser can already work completely offline:
- Offline maps
- Offline routing (BRouter)
- Offline search and points of interest

Just use the "Downloads" menu and download an offline (Mapsforge) map for your region.

---

There is also the Beta version (to be released the next days), that includes BRouter inside Cruiser,
so you do not need the BRouter app. And it automatically downloads offline maps + routing data:
- https://github.com/devemux86/cruiser/discussions/390

If you have more questions please let us know how we can help you. :slightly_smiling_face:

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 19:50 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> do you mean DEM with elevation
> data?
> And we still need a map to use dem files?
>
Yes,  election data.
We can use our DEM data using our mapforge map.

**Reply by devemux86** [reply]
> We can use our DEM data using our mapforge map.

This is something I want to add to Cruiser: reading digital elevation files.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 19:56 Emux, ***@***.***> wrote:

> We can use our DEM data using our mapforge map.
>
> This is something I want to add to Cruiser: reading digital elevation
> files.
>

It's good if you can add DEM menu so that we can use our own data.

**Comment by devemux86** [comment]
@yseo1110

They are a collection of text and xml files in UTF-8.
Many files have too little text to translate. :slightly_smiling_face: 

If you have questions or want to understand where a text is used, please let me know.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 21:21 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> Here are the Cruiser texts for translation: Cruiser.zip
> <https://github.com/devemux86/cruiser/files/13703579/Cruiser.zip>
>

How can read this text?
What kind of file is it.

**Reply by devemux86** [reply]
*.xml and *.properties are text files in UTF-8 encoding.

@yseo1110 you can read them with any file editor.

**Reply by devemux86** [reply]
@yseo1110 they contain the English texts for translation.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 21:30 Emux, ***@***.***> wrote:

> *.xml and *.properties are text files in UTF-8 encoding.
>

Sorry, I don,t have translation experience which made UTF8 encoding.
I think you can provide the menus for simple translation.

**Reply by devemux86** [reply]
@yseo1110 you just need to open each file with an editor and translate:
- In *.properties files what is after `=`
- In the 2 xml files it should be also easy

What operating system do you use, Windows?

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 21:54 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> you just need to open each file
> with an editor and translate:
>
>    - In *.properties files what is after =
>    - In the 2 xml files it should be also easy
>
> What operating system do you use, Windows?
>
No program to read utf8 on PC.
Just look after the text with phone.

**Reply by devemux86** [reply]
@yseo1110 

On Windows can use Notepad++: https://notepad-plus-plus.org

On Linux all editors can do it.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 22:07 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> On Windows can use Notepad++: https://notepad-plus-plus.org
>

Please find the attached photos for your reference.

On Linux all editors can do it.

**Reply by devemux86** [reply]
@yseo1110 there are not photos in the message.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 22:55 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> there are not photos in the
> message.
>
Send again here :)

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 22:07 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> On Windows can use Notepad++: https://notepad-plus-plus.org
>

I have notepad

**Comment by devemux86** [comment]
@yseo1110 there are no attachments in the messages.
Better use directly the GitHub website instead of email.

Also we will need text files, not images. :slightly_smiling_face: 

Why not downloading [Notepad++](https://notepad-plus-plus.org)? It helps with syntax highlighting etc.
Windows notepad can be very basic for most tasks.

**Comment by yseo1110** [comment]
On Mon, 18 Dec 2023, 23:13 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> there are no attachments in the
> messages.
> Better use directly the GitHub website instead of email.
>
> Also we will need text files, not images. 🙂
>
> Why not downloading Notepad++ <https://notepad-plus-plus.org>? It helps
> with syntax highlighting etc.
> Windows notepad can be very basic for most tasks.
>

Can I have only words for Crusier?

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 08:13 Youngil Seo, ***@***.***> wrote:

>
>
> On Mon, 18 Dec 2023, 23:13 Emux, ***@***.***> wrote:
>
>> @yseo1110 <https://github.com/yseo1110> there are no attachments in the
>> messages.
>> Better use directly the GitHub website instead of email.
>>
>> Also we will need text files, not images. 🙂
>>
>> Why not downloading Notepad++ <https://notepad-plus-plus.org>? It helps
>> with syntax highlighting etc.
>> Windows notepad can be very basic for most tasks.
>>
>
> Can I have only words for Crusier?
>
Can I have your personal email?

**Reply by devemux86** [reply]
@yseo1110 sure you can contact me via email.

The email is on the Cruiser page on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
("App support" section)

**Comment by yseo1110** [comment]
With regard to Korean translation, it is for desktop version, not App on
Google Play.

**Reply by devemux86** [reply]
@yseo1110 

Cruiser is developed as a single entity. So to include a new language we have to translate all the texts.

**Reply by devemux86** [reply]
@yseo1110 

It's **not** too much work if someone wants to have Cruiser in their language.
Many users have contributed new languages. :slightly_smiling_face: 

After the first translation by community, I take care of future translation needs.

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 16:37 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> The Cruiser is developed as a single entity. So to include a new language
> we have to translate all the texts.
>

May I have your nationality?  :)

**Reply by devemux86** [reply]
[Greece](https://en.wikipedia.org/wiki/Greece)

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 16:54 Emux, ***@***.***> wrote:

> Greece <https://en.wikipedia.org/wiki/Greece>
>
Oh, I see.
I think your timezone looks like European country :)
I'm Korean.

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 16:54 Emux, ***@***.***> wrote:

> Greece <https://en.wikipedia.org/wiki/Greece>
>

There are 35 items in Cruiser.
When I open Core,
#Button #Mark #Message #Point of interest

Should I translate the word after = ?

**Reply by devemux86** [reply]
@yseo1110 

The lines starting with `#` are comments, they do not need translation.

> Should I translate the word after = ?

Correctly, the other lines need translation in the text after the `=`.

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 17:28 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> The lines starting with # are comments, they do not need translation.
>
> Should I translate the word after = ?
>
> Correctly, the other lines need translation in the text after the =.
>

OK,  I understood :)
Thanks.
If any question,  I will let you know.

**Reply by devemux86** [reply]
@yseo1110 there are also 2 xml files, hope you can understand what to translate:

- `<item>Follow location</item>` the text "Follow location"
- `<string name="item_add_favorite">Add favorite</string>` the text "Add favorite"

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 17:49 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110> there are also 2 xml files, hope
> you can understand what to translate:
>
>    - <item>Follow location</item> the text "Follow location"
>    - <string name="item_add_favorite">Add favorite</string> the text "Add
>    favorite"
>
> It is not easy to collect the text for translate.
Of any questions,  I will ask you.

**Comment by devemux86** [comment]
@yseo1110 

When you finish a file, please send it to me so I can check if it works correctly inside Cruiser.

You can attach it here or send me an email (it is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser)).

**Comment by yseo1110** [comment]
On Tue, 19 Dec 2023, 18:56 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> When you finish a file, please send it to me so I can check if it works
> correctly inside Cruiser.
>
OK .

> You can attach it here or send me an email (it is on Google Play
> <https://play.google.com/store/apps/details?id=gr.talent.cruiser>).
>
Yes, confirmed your private email.

**Comment by devemux86** [comment]
Cruiser in Korean preview thanks to work by @yseo1110.
Korean voice navigation is also available.

<img src="https://github.com/devemux86/cruiser/assets/3484020/edd9b80a-bebe-4301-bae0-7270876406df" width="50%"/>

![Desktop](https://github.com/devemux86/cruiser/assets/3484020/26f0fc29-cdc9-4adb-99c4-ede1f32de064)

**Comment by devemux86** [comment, marked as answer]
Korean translation and voice navigation is available in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

@yseo1110 thanks for the translation! :+1:

**Comment by yseo1110** [comment]
On Tue, 26 Dec 2023, 21:47 Emux, ***@***.***> wrote:

> Korean translation and voice navigation is available in Cruiser 3.0.37
> <https://play.google.com/store/apps/details?id=gr.talent.cruiser>.
>

Very nice.
Hope many Korean use Cruiser App. :)

Please do not hesitate to contact me if you need further assistance.

Youngil Seo

**Comment by yseo1110** [comment]
On Tue, 26 Dec 2023, 21:47 Emux, ***@***.***> wrote:

> Korean translation and voice navigation is available in Cruiser 3.0.37
> <https://play.google.com/store/apps/details?id=gr.talent.cruiser>.
>
> @yseo1110 <https://github.com/yseo1110> thanks for the translation! 👍
>
>
If you can add Brouter offline menu, it will be much appreciated.
Also required add DEM and import photo so that we can see photo waypoint.

Please apply Korea language to CRUSIER 1.4 desktop.

**Comment by yseo1110** [comment]
Dear Emux

Please find the attached sent msg to you and waiting for your reply.


Youngil

**Reply by devemux86** [reply]
@yseo1110 there is no message attached.

**Comment by yseo1110** [comment]
Dear Emux

If it is impossible to add  Brouter offline function to latest version,
please apply Korean language to 1.4 desktop.

Many thanks for your kind efforts.


Youngil Seo

**Reply by devemux86** [reply]
@yseo1110 thanks for the suggestion.

BRouter (offline) routing is only available on Android.

On desktop it is better to use the [BRouter website](https://brouter.de/brouter-web/) or if using custom .rd5 files run the [BRouter web client](https://github.com/nrenner/brouter-web).

About Cruiser, development is taking place on the latest version. :slightly_smiling_face:

**Comment by yseo1110** [comment]
Dear Emux

When we faced problem to read Korean from desktop Crusier, you infome me
the way to appear Korean language correctly.
What it was?
Can you send me again?


Youngil

**Reply by devemux86** [reply]
@yseo1110 

After changing the language, it is required to restart the app.
(there is also a message "Restart required")

**Comment by yseo1110** [comment]
On Thu, 11 Jan 2024, 16:48 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> After changing the language, it is required to restart the app.
> (there is also a message "Restart required")
>

Can you give me the link that sent me previously, maybe regarding Java.

**Reply by devemux86** [reply]
@yseo1110 

OpenJDK: https://adoptium.net

**Comment by yseo1110** [comment]
On Thu, 11 Jan 2024, 17:09 Emux, ***@***.***> wrote:

> @yseo1110 <https://github.com/yseo1110>
>
> OpenJDK: https://adoptium.net
>

Thank you very much.

**Comment by Petak73** [comment]
@devemux86 I am willing to translate it into Bulgarian. Which encoding should be used for Cyrillic or it doesn't matter? I will be using Notepad

**Reply by devemux86** [reply]
@Petak73 Thanks for the help.

If you are interested, please send me an email, so I can send you the new texts for translation.
(the above texts are very old and have been changed many times)

The text files must have the **UTF-8** encoding.

**Comment by Petak73** [comment]
Hi,
Please send the most current English files to this email. I have already started - see the file attached. Am I doing this right?
Cheers,
Peter
________________________________

**Reply by devemux86** [reply]
@Petak73 

Please do not start anything, as these files are very old and have been changed many times and cannot be used!

Send me first an email, it can be seen on [Google Play page of Cruiser](https://play.google.com/store/apps/details?id=gr.talent.cruiser), so I can send you the correct files for translation.

**Comment by devemux86** [comment]
@Petak73 Bulgarian voice language is available in **Cruiser 5.9.15**:

- https://github.com/devemux86/cruiser/discussions/901#discussioncomment-18363595
